pact4s/src/test/scala/org/broadinstitute/dsde/rawls/provider/RawlsProviderSpec.scala

```diff
@@ -0,0 +1,321 @@
+
+  private val providerStatesHandler: StateManagementFunction = StateManagementFunction {
+    case ProviderState(States.rawlsOK, _) =>
+      mockSubsystemsStatus(true)
+    case ProviderState(States.oneSnapshot, _) =>
+      mockEnumerateSnapshots(
+        mockSnapshotServiceConstructor(RawlsRequestContext(userInfo = mockUserInfo, otelContext = mockOtelContext)),
+        mockedEnumerateSnapshotsResponse
+      )
+    case ProviderState(States.snapshotCreatePolicy, _) =>
+      mockCreateSnapshot(
+        mockSnapshotServiceConstructor(RawlsRequestContext(userInfo = mockUserInfo, otelContext = mockOtelContext)),
+        dataRepoSnapshotResource
+      )
+    case _ =>
+      loggerIO.debug("State not found")
+  }
+
+  private def mockEnumerateSnapshots(mockSnapshotService: SnapshotService,
+                                     mockResponse: SnapshotListResponse
+  ): OngoingStubbing[Future[SnapshotListResponse]] =
+    when {
+      mockSnapshotService.enumerateSnapshotsById(anyString(), anyInt(), anyInt(), any[Option[UUID]])
+    } thenReturn {
+      Future.successful(mockResponse)
+    }
+
+  private def mockCreateSnapshot(mockSnapshotService: SnapshotService,
+                                 mockResponse: DataRepoSnapshotResource
+  ): OngoingStubbing[Future[DataRepoSnapshotResource]] =
+    when {
+      mockSnapshotService.createSnapshotByWorkspaceId(anyString(), any[NamedDataRepoSnapshot])
+    } thenReturn {
+      Future.successful(mockResponse)
+    }
+
+  lazy val pactBrokerUrl: String = sys.env.getOrElse("PACT_BROKER_URL", "")
+  lazy val pactBrokerUser: String = sys.env.getOrElse("PACT_BROKER_USERNAME", "")
+  lazy val pactBrokerPass: String = sys.env.getOrElse("PACT_BROKER_PASSWORD", "")
+  // Provider branch, semver
+  lazy val providerBranch: String = sys.env.getOrElse("PROVIDER_BRANCH", "")
+  lazy val providerVer: String = sys.env.getOrElse("PROVIDER_VERSION", "")
+  // Consumer name, branch, semver (used for webhook events only)
+  lazy val consumerName: Option[String] = sys.env.get("CONSUMER_NAME")
+  lazy val consumerBranch: Option[String] = sys.env.get("CONSUMER_BRANCH")
+  // This matches the latest commit of the consumer branch that triggered the webhook event
+  lazy val consumerVer: Option[String] = sys.env.get("CONSUMER_VERSION")
+
+  var consumerVersionSelectors: ConsumerVersionSelectors = ConsumerVersionSelectors()
+  // consumerVersionSelectors = consumerVersionSelectors.mainBranch
+  // The following match condition basically says
+  // 1. If verification is triggered by consumer pact change, verify only the changed pact.
+  // 2. For normal Sam PR, verify all consumer pacts in Pact Broker labelled with a deployed environment (alpha, dev, prod, staging).
+  consumerBranch match {
+    case Some(s) if !s.isBlank => consumerVersionSelectors = consumerVersionSelectors.branch(s, consumerName)
+    case _                     => consumerVersionSelectors = consumerVersionSelectors.deployedOrReleased.mainBranch
```

Comment:
@calypsomatic At this point, the new `wds-rawls` pact still belongs to the `aj-1697-rawls-contract` branch of the `terra-workspace-data-service` repository and has not yet been merged or deployed. As a result, `consumerVersionSelectors.deployedOrReleased.mainBranch` does not yield any results. Try directly referencing the `wds` branch `aj-1697-rawls-contract` for the provider test to locate the pact from the pact broker. This may resolve the error you saw during the provider test.
```suggestion
    case _                     => consumerVersionSelectors = consumerVersionSelectors.deployedOrReleased.mainBranch.branch("aj-1697-rawls-contract", Some("wds"))
```